app/uk/gov/hmrc/gform/validation/AddressValidation.scala

```diff
@@ -186,4 +187,18 @@ class AddressValidation[D <: DataOrigin](implicit messages: Messages, sse: Smart
         mkErrors(fieldValue, atomicFcId)(messageKey, vars)
       case _ => validationSuccess
     }
+
+  private def checkPostcode(
+    fieldValue: FormComponent,
+    atomicFcId: ModelComponentId.Atomic,
+    messageKey: String
+  )(
+    xs: Seq[String]
+  ): ValidatedType[Unit] =
+    xs.filterNot(_.isEmpty) match {
+      case value :: Nil if !PostcodeLookupValidation.checkPostcode(value) =>
```

Comment:
Should we extract validation logic to some utility class?

Reply:
`lengthLimitValidation ` and `checkPostcode` differs only on predicate we should refactor that to have single implementation, but private method sounds good enough. But if you can see that this pattern is repeated elsewhere than fell free to do broader refactoring using utility class